app/gui/view/graph-editor/src/lib.rs

```diff
@@ -1429,11 +1428,13 @@ impl GraphEditorModelWithNetwork {
         let position: Vector2 = match way {
             AddNodeEvent => default(),
             StartCreationEvent | ClickingButton if selection.is_some() =>
-                self.find_free_place_under(selection.unwrap()),
-            StartCreationEvent => mouse_position,
+                new_node_position::under(self, selection.unwrap()),
+            StartCreationEvent =>
+                new_node_position::at_mouse_aligned_to_close_nodes(self, mouse_position),
             ClickingButton =>
-                self.find_free_place_for_node(screen_center, Vector2(0.0, -1.0)).unwrap(),
-            DroppingEdge { .. } => mouse_position,
+                new_node_position::on_ray(self, screen_center, Vector2(0.0, -1.0)).unwrap(),
+            DroppingEdge { edge_id } =>
+                new_node_position::at_mouse_aligned_to_source_node(self, edge_id, mouse_position),
         };
```

Comment:
The separate module made it look so nice! Perhaps this whole match should also be refactored to function inside `new_node_position`.